node/api/api.go

```diff
@@ -29,6 +32,25 @@ type Error struct {
 	// be valid or invalid depending on the current state of a module.
 }
 
+type PaginationRequest struct {
+	HasPaginationQuery     bool
+	PaginationQueryIsValid bool
+	Start                  int
+	Limit                  int
+}
+
+type PaginationResponse struct {
+	Start      int `json:"start"`
+	Limit      int `json:"limit"`
+	TotalPages int `json:"total_pages"`
+}
+
+type PaginationWrapper struct {
+	Start      int
+	End        int
+	TotalPages int
+}
+
```

Comment:
There are probably enough new pagination types and methods to move the generic ones into a `node/api/pagination.go` file instead of having them in `api.go`.
The ones that are specific to a module can still be in the corresponding `[module].go` file.